src/abstractdataframe/selection.jl

```diff
@@ -351,8 +353,8 @@ function _transformation_helper(df::AbstractDataFrame, col_idx::AbstractVector{I
     end
 end
 
-function _gen_colnames(@nospecialize(res), newname::Union{AbstractVector{Symbol},
-                                                          Type{AsTable}, Nothing})
+function _gen_colnames(res, newname::Union{AbstractVector{Symbol},
+                                           Type{AsTable}, Nothing})
```

Comment:
So here you think it's better to specialize? :-D

Reply:
The benchmarks show that it is minimally better not to specialize given our test set :) - these things are super strange. However the difference is almost imperceptible, so I reverted `@nospecialize` as probably in practical loads not-specializing here might be better.